chain/messagesigner/messagesigner_consensus.go

```diff
@@ -0,0 +1,99 @@
+package messagesigner
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	libp2pconsensus "github.com/libp2p/go-libp2p-consensus"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	consensus "github.com/filecoin-project/lotus/lib/consensus/raft"
+	"github.com/filecoin-project/lotus/node/modules/dtypes"
+)
+
+type MessageSignerConsensus struct {
+	MsgSigner
+	consensus *consensus.Consensus
+}
+
+func NewMessageSignerConsensus(
+	wallet api.Wallet,
+	mpool MpoolNonceAPI,
+	ds dtypes.MetadataDS,
+	consensus *consensus.Consensus) *MessageSignerConsensus {
+
+	ds = namespace.Wrap(ds, datastore.NewKey("/message-signer-consensus/"))
+	return &MessageSignerConsensus{
+		MsgSigner: &MessageSigner{
+			wallet: wallet,
+			mpool:  mpool,
+			ds:     ds,
+		},
+		consensus: consensus,
+	}
+}
+
+func (ms *MessageSignerConsensus) IsLeader(ctx context.Context) bool {
+	return ms.consensus.IsLeader(ctx)
+}
+
+func (ms *MessageSignerConsensus) RedirectToLeader(ctx context.Context, method string, arg interface{}, ret interface{}) (bool, error) {
+	ok, err := ms.consensus.RedirectToLeader(method, arg, ret.(*types.SignedMessage))
+	if err != nil {
+		return ok, err
+	}
+	return ok, nil
+}
+
+func (ms *MessageSignerConsensus) SignMessage(
+	ctx context.Context,
+	msg *types.Message,
+	spec *api.MessageSendSpec,
+	cb func(*types.SignedMessage) error) (*types.SignedMessage, error) {
+
+	signedMsg, err := ms.MsgSigner.SignMessage(ctx, msg, spec, cb)
+	if err != nil {
+		return nil, err
+	}
+
+	// We can't have an empty/default uuid as part of the consensus state so generate a new uuid if spec is empty
+	u := uuid.New()
+	if spec != nil {
+		u = spec.MsgUuid
```

Comment:
This needs to be set here because the consensus state needs a non empty uuid. And the API today allows you to push a message with an empty uuid
Are you saying you want to set a new value elsewhere instead?

Reply:
It can't be a 0 value because then we would reject any messages that don't have it originally set

Reply:
But isn't this creating a regression for the fix in https://github.com/filecoin-project/lotus/pull/9303